plugins/modules/ec2_imagebuilder_component.py

```diff
@@ -0,0 +1,399 @@
+def construct_create_params(module):
+    params = dict()
+
+    # required params
+    params['name'] = module.params.get('name')
+    params['platform'] = module.params.get('platform')
+    params['semantic_version'] = module.params.get('semantic_version')
+
+    # optional params
+    optional_params_keys = [
+        'change_description',
+        'data',
+        'description',
+        'kms_key_id',
+        'uri',
+        'tags']
+
+    for k in optional_params_keys:
+        if module.params.get(k) is not None:
+            params[k] = module.params.get(k)
+
+    # ensure params are in camel case for compatibility with EC2 Image Builder API
+    params = snake_dict_to_camel_dict(params)
+
+    return params
+
+
+def create_or_tag_component(connection, module, arn, current_component_params):
+    params = construct_create_params(module)
+
+    changed = False
+
+    # attempt to create component if it does not exist or its parameters have changed
+    if not current_component_params or has_component_params_changed(
+            params, current_component_params):
+        connection.create_component(**params)
+        changed = True
+
+    # update tags if component exists and tags have changed
+    if current_component_params and have_component_tags_changed(params, current_component_params):
+        update_component_tags(connection, arn, params, current_component_params)
+        changed = True
+
+    # if a change has been made, refetch
+    if changed:
+        current_component_params = get_component(connection, module, arn)
+
+    module.exit_json(changed=changed, component=camel_dict_to_snake_dict(current_component_params))
+
+
+def delete_component(connection, module, arn, component):
+    """
+    Delete an EC2 Image Builder component if it exists
+
+    :param connection: AWS boto3 imagebuilder connection
+    :param module: Ansible module
+    :param component: a dict of EC2 Image Builder component representing current state of resource, or None
+    :return: True if component was deleted else False
+    """
+
+    changed = False
+
+    if component:
+        try:
+            connection.delete_component(
+                componentBuildVersionArn=arn)
+            changed = True
+        except (BotoCoreError, ClientError) as e:
+            module.fail_json_aws(e)
+
+    module.exit_json(changed=changed, component=dict(arn=arn))
+
+
+def get_aws_account_id(module):
+    return module.client('sts').get_caller_identity()['Account']
+
+
+def get_component(connection, module, arn):
+    """
+    Get an EC2 Image Builder component by ARN. If not found, return None.
+
+    :param connection: AWS boto3 imagebuilder connection
+    :param module: Ansible module
+    :param arn: ARN of Image Builder component to get
+    :return: boto3 Image Builder component dict or None if not found
+    """
+
+    try:
+        return connection.get_component(componentBuildVersionArn=arn)['component']
+    except (BotoCoreError, ClientError) as e:
```

Comment:
We have a helper which I prefer to use here:

```suggestion
    except is_boto3_error_code('ResourceNotFoundException'):
        return None
    except (BotoCoreError, ClientError) as e:
        module.fail_json_aws(e)
```